src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchProfilesProjectItemProvider.cs

```diff
@@ -0,0 +1,358 @@
+                    projectItems.Add(projectItem);
+                }
+            }
+
+            return projectItems;
+        }
+
+        /// <remarks>
+        /// Launch profiles do not represent a file on disk, so adding one via a path is
+        /// a meaningless operation.
+        /// </remarks>
+        public Task<IProjectItem> AddAsync(string path)
+        {
+            throw new InvalidOperationException($"The {nameof(LaunchProfilesProjectItemProvider)} does not support adding items as paths.");
+        }
+
+        /// <remarks>
+        /// Launch profiles do not represent a file on disk, so adding one via a path is
+        /// a meaningless operation.
+        /// </remarks>
+        public Task<IReadOnlyList<IProjectItem>> AddAsync(IEnumerable<string> paths)
+        {
+            throw new InvalidOperationException($"The {nameof(LaunchProfilesProjectItemProvider)} does not support adding items as paths.");
+        }
+
+        public async Task<IProjectItem?> FindItemByNameAsync(string evaluatedInclude)
+        {
+            IEnumerable<IProjectItem> items = await GetItemsAsync();
+            return items.FirstOrDefault(item => StringComparers.ItemNames.Equals(item.EvaluatedInclude, evaluatedInclude));
+        }
+
+        public async Task<IImmutableSet<string>> GetExistingItemTypesAsync()
+        {
+            ILaunchSettings? snapshot = await _launchSettingsProvider.WaitForFirstSnapshot(Timeout.Infinite);
+            Assumes.NotNull(snapshot);
+
+            return snapshot.Profiles.Count > 0
+                ? s_itemTypes
+                : ImmutableSortedSet<string>.Empty;
+        }
+
+        public async Task<IProjectItem?> GetItemAsync(IProjectPropertiesContext context)
+        {
+            if (context.ItemType is not null
+                && !StringComparers.ItemTypes.Equals(ItemType, context.ItemType))
+            {
+                return null;
+            }
+
+            IEnumerable<IProjectItem> items = await GetItemsAsync();
+            return items.FirstOrDefault(item => StringComparers.ItemNames.Equals(item.EvaluatedInclude, context.ItemName));
+        }
+
+        public async Task<IEnumerable<IProjectItem>> GetItemsAsync()
+        {
+            ILaunchSettings? snapshot = await _launchSettingsProvider.WaitForFirstSnapshot(Timeout.Infinite);
+            Assumes.NotNull(snapshot);
+
+            if (snapshot.Profiles.Count == 0)
+            {
+                return ImmutableArray<IProjectItem>.Empty;
+            }
+
+            return snapshot.Profiles.Select(p => new ProjectItem(p.Name ?? string.Empty, _project.FullPath, this));
+        }
+
+        public Task<IEnumerable<IProjectItem>> GetItemsAsync(string itemType)
+        {
+            if (StringComparers.ItemTypes.Equals(itemType, ItemType))
+            {
+                return GetItemsAsync();
+            }
+
+            return Task.FromResult<IEnumerable<IProjectItem>>(ImmutableArray<ProjectItem>.Empty);
```

Comment:
Ditto re boxing:

```suggestion
            return Task.FromResult(Enumerable.Empty<ProjectItem>());
```

If this is called a lot you can cache the task too. Maybe even on our `TaskResult` type, similar to `TaskResult.Null()<T>`.